### PR TITLE
Add info icons

### LINKED ISSUE #172: Improvements of the GUI: Add info boxes
## User story
As a user
I want / need an Intuitiv and understandable GUI
So that I can do the tasks easily and fastly.  

## Acceptance criteria
* There should be info buttons, which open a small info window showing short, but beneficial instruction and provide more information about the underlying functionality, e.g. 
Info_button

Following text should the info buttons show:
- Hyperparameter: The hyperparameter are settings to adjust the distinction between added and deleted frames and a pixel difference within a frame.
- gap open: (add here explanation for gap open)
- gap extended: (add here explanation for gap extended)
- upload mask: Upload a png with white and black rectangles. Everything that is within white rectangles in video will be considered in the video difference computation and everything that is within black rectangles will be not considered in the video difference computation.

## Definition of done (DoD)
* Create feature branch
* Implement the functionality
* Opened pull request and assigned reviewer
* If necessary: required changes are implemented
* Delete feature branch
* If necessary: Component Test has been written
* Code review has been completed and code has been merged

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

### PR DESCRIPTION
**Pull request by zino212**
Closes #172 

This PR adds info icons in the settings screen. When hovering the icons, a corresponding text is displayed.

### CODE REVIEW

**Comment by zino212** [comment]
In my opinion, the at least the settings screen needs to be refactored. There are some adjustments to be made but they are not part of this ticket.
 The sizes and scales of the various components are not optimized.

File: GUI/src/main/kotlin/ui/screens/SettingsScreen.kt [hunk 1]
```diff
@@ -80,3 +111,62 @@ fun RowScope.SaveButton(state: MutableState<AppState>) {
...
+
+    Box(
+        modifier =
+            Modifier
+                .onPointerEvent(PointerEventType.Enter) {
+                    isHovered = true
+                }
+                .onPointerEvent(PointerEventType.Exit) {
+                    isHovered = false
+                },
```
Comment (a-miscellaneous) [inline GUI/src/main/kotlin/ui/screens/SettingsScreen.kt:128]: onliners are more readable imo:
.onPointerEvent(PointerEventType.Enter) { isHovered = true }

File: GUI/src/main/kotlin/ui/screens/SettingsScreen.kt [hunk 2]
```diff
@@ -80,3 +111,62 @@ fun RowScope.SaveButton(state: MutableState<AppState>) {
...
+                    MaterialTheme.colors.primary
+                } else {
+                    MaterialTheme.colors.onBackground.copy(
+                        alpha = LocalContentAlpha.current,
+                    )
+                },
+            modifier =
+                Modifier
+                    .size(24.dp),
+        )
```
Comment (a-miscellaneous) [inline GUI/src/main/kotlin/ui/screens/SettingsScreen.kt:144]: one-line

**Review by a-miscellaneous — CHANGES_REQUESTED** [review]
minor changes.
the styling needs to be fixed, probably part of another ticket?

**Comment by akriese** [comment]
I fixed the space between titles and tooltip icons, but the tooltips for buttons are still an issue. Will look into that tomorrow :)

**Comment by akriese** [comment]
rebased on most recent main

File: GUI/src/main/kotlin/ui/screens/SettingsScreen.kt [hunk 3]
```diff
@@ -2,40 +2,55 @@ package ui.screens
...
+    val textForGapExtended =
+        "Penalty score for extending an existing gap (insertion/deletion) in the alignment.\n" +
+            "Increasing this value will favor longer gaps in the alignment.\n" +
+            "The gapOpenPenalty should be smaller than the gapExtensionPenalty.\n" +
+            "If both parameters are equal, the algorithm will behave like the Needleman-Wunsch algorithm,\n" +
+            "meaning that new gaps are as likely as extending existing gaps."
+    val textForMask =
+        "Upload a png with white and black rectangles.\n" +
+            "The area marked with white rectangles will be included in the video difference computation \n" +
+            "and the area with black rectangles will not be included in the computation."
```
Comment (a-miscellaneous) [inline GUI/src/main/kotlin/ui/screens/SettingsScreen.kt:39]: technically wrong:
Upload a png with a clear background and colored areas.\n" +
            "The clear areas will be included in the video difference computation \n" +
            "and the area with colored in will not be included in the computation

**Review by a-miscellaneous — CHANGES_REQUESTED** [review]
i don't see tooltips?
image

**Comment by akriese** [comment]
rebased after merge of #184

**Review by a-miscellaneous — APPROVED** [review]
LGTM

**Pull request merged**